sdks/java/core/src/main/java/org/apache/beam/sdk/io/WriteFiles.java

```diff
@@ -133,37 +133,50 @@
   private final ValueProvider<Integer> numShardsProvider;
   private final boolean windowedWrites;
   private int maxNumWritersPerBundle;
+  // If set,this is used instead of DynamicDestinations.getSideInputs. This is used by the Beam
+  // translation code.
+  private final List<PCollectionView<?>> overriddenSideInputs;
```

Comment:
Mark this `@Nullable` [if you leave it as-is](https://en.wikipedia.org/wiki/Tony_Hoare#Apologies_and_retractions) but actually please don't. You can set it in the constructor either way, yes?